src/layout/GenericComponent.tsx

```diff
@@ -89,6 +90,18 @@ function ActualGenericComponent<Type extends CompTypes = CompTypes>({
   const isValid = !hasValidationErrors(validations);
   const isHidden = useIsHiddenComponent();
 
+  const { searchParams, clearSearchParam } = useNavigationParams();
+  const componentId = searchParams.get(SearchParams.FocusComponentId);
+  const focusNode = useResolvedNode(componentId);
+  const navigateTo = useNavigateToNode();
+
+  React.useEffect(() => {
+    if (focusNode != null) {
+      navigateTo(focusNode);
+    }
+    clearSearchParam(SearchParams.FocusComponentId);
+  }, [componentId, clearSearchParam, navigateTo, focusNode]);
```

Comment:
Why does every component have to do this? :thinking: Doesn't that mean that when rendering the page, N components (possibly hundreds) will call `navigateTo(...)` and clear the search params? I would think this functionality is better placed in `Form.tsx`, as it's a peculiarity of a form, not for each component.

Reply:
You're right! I'll see if I get it to work in Form.tsx